src/model/CoilCoolingDX.cpp

```diff
@@ -144,11 +156,70 @@ namespace model {
         }
       }
 
+      // CoilSystemCoolingDXHeatExchangerAssisted
+      {
+        auto coilSystems = this->model().getConcreteModelObjects<CoilSystemCoolingDXHeatExchangerAssisted>();
+        for (const auto& coilSystem : coilSystems) {
+          if (coilSystem.coolingCoil().handle() == this->handle()) {
+            return coilSystem;
+          }
+        }
+      }
```

Comment:
Update `containingHVACComponent` as we can now have `Coil:Cooling:DX` on `CoilSystem:Cooling:DX:HeatExchangerAssisted`.